CLI-27: When the --no-interaction flag is passed, we should either fail gracefully or force interacting with the user when required

Example with interaction:

```
acli ssh-key:create

 Please enter a filename for your new local SSH key. Press enter to use default value [id_rsa_acquia]:
 >

 Enter a password for your SSH key:
 >


 [ERROR] This value is too short. It should have 5 characters or more.


 Enter a password for your SSH key:
 >
```

And without:

```
acli -n ssh-key:create
PHP Fatal error:  Uncaught TypeError: Acquia\Cli\Command\Ssh\SshKeyCreateCommand::determinePassword(): Return value must be of type string, null returned in phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php:145
Stack trace:
#0 phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php(57): Acquia\Cli\Command\Ssh\SshKeyCreateCommand->determinePassword(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php(41): Acquia\Cli\Command\Ssh\SshKeyCreateCommand->createSshKey(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 phar:///usr/local/bin/acli/vendor/symfony/console/Command/Command.php(299): Acquia\Cli\Command\Ssh\SshKeyCreateCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 phar:///usr/local/bin/acli/src/Command/CommandBase.php(409): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(996): Acquia\Cli\Command\CommandBase->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(295): Symfony\Component\Console\Application->doRunCommand(Object(Acquia\Cli\Command\Ssh\SshKeyCreateCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///usr/local/bin/acli/bin/acli(86): Symfony\Component\Console\Application->run()
#8 /usr/local/bin/acli(14): require('phar:///usr/loc...')
#9 {main}
  thrown in phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php on line 145

Fatal error: Uncaught TypeError: Acquia\Cli\Command\Ssh\SshKeyCreateCommand::determinePassword(): Return value must be of type string, null returned in phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php:145
Stack trace:
#0 phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php(57): Acquia\Cli\Command\Ssh\SshKeyCreateCommand->determinePassword(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php(41): Acquia\Cli\Command\Ssh\SshKeyCreateCommand->createSshKey(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 phar:///usr/local/bin/acli/vendor/symfony/console/Command/Command.php(299): Acquia\Cli\Command\Ssh\SshKeyCreateCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 phar:///usr/local/bin/acli/src/Command/CommandBase.php(409): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(996): Acquia\Cli\Command\CommandBase->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(295): Symfony\Component\Console\Application->doRunCommand(Object(Acquia\Cli\Command\Ssh\SshKeyCreateCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///usr/local/bin/acli/bin/acli(86): Symfony\Component\Console\Application->run()
#8 /usr/local/bin/acli(14): require('phar:///usr/loc...')
#9 {main}
  thrown in phar:///usr/local/bin/acli/src/Command/Ssh/SshKeyCreateCommand.php on line 145
```

--- a/src/Command/Ssh/SshKeyCreateCommand.php
+++ b/src/Command/Ssh/SshKeyCreateCommand.php
@@ -162,18 +162,19 @@ class SshKeyCreateCommand extends SshKeyCommandBase {
     if ($input->getOption('password')) {
       $password = $input->getOption('password');
       $this->validatePassword($password);
+      return $password;
     }
-    else {
+    if ($input->isInteractive()) {
       $question = new Question('Enter a password for your SSH key');
       $question->setHidden($this->localMachineHelper->useTty());
       $question->setNormalizer(static function ($value) {
         return $value ? trim($value) : '';
       });
       $question->setValidator(Closure::fromCallable([$this, 'validatePassword']));
-      $password = $this->io->askQuestion($question);
+      return $this->io->askQuestion($question);
     }
 
-    return $password;
+    throw new AcquiaCliException('Could not determine the SSH key password. Either use the --password option or else run this command in an interactive shell.');
   }
 
   /**

